File: 1101-1200/1161-Maximum Level Sum of a Binary Tree/1161-Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int maxSum = root->val, maxLevel = 1, level = 0;
        queue<TreeNode*> Q;
        Q.push(root);
        while (!Q.empty()) {
            int qs = Q.size(), s = 0;
            for (int i = 0; i < qs; ++i) {
                TreeNode* node = Q.front();
                Q.pop();
                s += node->val;
                if (node->left) {
                    Q.push(node->left);
                }
                if (node->right) {
                    Q.push(node->right);
                }
            }
            ++level;
            if (s > maxSum) {
                maxSum = s;
                maxLevel = level;
            }
        }

        return maxLevel;
    }
};

File: 1101-1200/1161-Maximum Level Sum of a Binary Tree/1161-Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

import collections
class Solution:
    def maxLevelSum(self, root: TreeNode) -> int:
        maxSum = root.val
        maxLeval = 1
        Q = collections.deque([root])
        level = 0
        while Q:
            s = 0
            for _ in range(len(Q)):
                node = Q.popleft()
                s += node.val
                if node.left:
                    Q.append(node.left)
                if node.right:
                    Q.append(node.right)
            
            level += 1
            if s > maxSum:
                maxSum = s
                maxLeval = level
        
        return maxLeval
